scapy/layers/dot11.py

```diff
@@ -1282,7 +1310,7 @@ class Dot11EltCountry(Dot11Elt):
         # When this extension is last, padding appears to be omitted
         ConditionalField(
             ByteField("pad", 0),
-            lambda pkt: (len(pkt.descriptors) + 1) % 2
+            country_info_element_padding_condition
```

Comment:
I'd do something simpler in one line
```
lambda pkt: ((len(pkt.descriptors) + 1) % 2) if pkt.len is None else (pkt.len % 3)
```
just to handle both dissecting and build cases. Wouldn't that work?

Reply:
I added the `is None` test for the build case.
I think the long version is more readable because it explains to the reader the logic behind the condition, and it also explicitly fails for scenarios such as `pkt.len == 83`. In my version, a `ValueError` is expected, but in your version, the dissector will assume there's a single padding byte, even though it is wrong (because if the length is 83 and there's a padding byte, then it is 82, but 82 isn't divided by 3). At the moment, we don't know if there's a case with % == 2 (e.g., 83), so your version also works.

Reply:
Isn't the padding byte included in the length?

Reply:
The padding byte is indeed included, it can 83 if the transmitter is non-compliant. This PR started because we found "in the wild" a non-compliant transmitter, using a length of 81, not adding the required padding byte (by the length).
I'll change to a one-liner, but the cost is that handling of non-compliant transmitters is worse.